.github/workflows/php-lint.yml

```diff
@@ -0,0 +1,59 @@
+name: PHP Code Linting
+
+on:
+  push:
+    branches:
+      - trunk
+      - 'feature/**'
+      - 'release/**'
+    # Only run if PHP-related files changed.
+    paths:
+      - '.github/workflows/php-lint.yml'
+      - '**.php'
+      - 'phpcs.xml.dist'
+      - 'phpstan.neon.dist'
+      - 'composer.json'
+      - 'composer.lock'
+  pull_request:
+    # Only run if PHP-related files changed.
+    paths:
```

Comment:
Somewhat related to https://github.com/WordPress/php-ai-client/pull/22/files#r2205986997, in other WordPress organization projects we also have these same branch restrictions under `pull_request`, to ensure that only pull requests _against_ those branches trigger the workflows.

I'm not strongly in favor of either way, but might be worth aligning. If you create some quick branch just to merge into some other secndary branch that still will require review anyway, I'd argue that running CI isn't needed.

Reply:
Works for me 👍